include/circular_buffer.hpp

```diff
@@ -246,17 +246,72 @@ class circular_buffer
         size_type                   _size;
         ClaimType*                  _claim_strategy;
 
+    class WriterToken {
+        buffer_writer<U>& _parent;
+        std::size_t  _index = 0;
+        std::size_t  _n_slots_to_claim = 0;
+        std::int64_t _offset = 0;
+        bool         _published_data = false;
+    public:
+    explicit WriterToken(buffer_writer<U>& parent) noexcept : _parent(parent) {};
+    explicit constexpr WriterToken(buffer_writer<U>& parent, std::size_t index, std::int64_t sequence, std::size_t n_slots_to_claim) noexcept :
+        _parent(parent), _index(index), _n_slots_to_claim(n_slots_to_claim), _offset(sequence - n_slots_to_claim) { }
+    WriterToken(const WriterToken&) = delete;
+    WriterToken& operator=(const WriterToken&) = delete;
+    explicit WriterToken(WriterToken&& other) noexcept : WriterToken(other._parent) {
+        *this = std::move(other);
+    };
+    WriterToken& operator=(WriterToken&& other) noexcept {
+        std::swap(_parent, other._parent);
+        std::swap(_index, other._index);
+        std::swap(_n_slots_to_claim, other._n_slots_to_claim);
+        std::swap(_offset, other._offset);
+        std::swap(_published_data, other._published_data);
+        other._n_slots_to_claim = 0;
+        other._published_data = true;
+        return *this;
+    };
+    ~WriterToken() {
+        if constexpr (std::is_base_of_v<MultiThreadedStrategy<SIZE, WAIT_STRATEGY>, ClaimType>) {
+            if (_n_slots_to_claim) {
+                throw std::runtime_error(fmt::format("circular_buffer::multiple_writer::WriterToken() - did not publish {} samples", _n_slots_to_claim));
```

Comment:
Throwing in a destructor is not a good idea

Reply:
So is not calling the publish function on this RAII method ... any other suggestions how to indicate this dramatic API usage failure? Derek suggested that he wouldn't mind if the house would burn down ... a bit too dramatic for me but we need something in between.

Reply:
API usage failure could be `std::terminate` IMO - exceptions can be handled and ignored, bad API usage should not be easy to ignore.